pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -512,9 +515,22 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 
 	// we compute the minimum ordinal of the target sequence for a destructive update based on the strategy.
 	updateMin := 0
+	maxUnavailable := 1
 	if set.Spec.UpdateStrategy.RollingUpdate != nil {
 		updateMin = int(*set.Spec.UpdateStrategy.RollingUpdate.Partition)
+		maxUnavailable, err = intstrutil.GetValueFromIntOrPercent(intstrutil.ValueOrDefault(set.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable, intstrutil.FromInt(1)), int(replicaCount), false)
+		if err != nil {
+			return &status, err
+		}
+		// if the feature was enabled and then later disabled, MaxUnavailable may have a value
+		// other than 1. Reset maxUnavailable to 1 to revert to the expected behavior when this
+		// feature gate is not enabled.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.MaxUnavailableStatefulSet) {
+			maxUnavailable = 1
```

Comment:
@lavalamp  @kow3ns added this extra logic for the case when feature is enabled and then disabled again. ptal

Reply:
kind of weird that you first parse the value and then check for the feature gate.
Prefer to do one or the other. That way, if somehow there is a bug in the validation or parsing, administrators can still disable the feature and have everything working as if the field was not there.